Adds Eslint and formFiller component

pzubar opened the pull request:
1. Eslint and prettier were added
2. formEditor became statefull component
3. formFiller component development started

pzubar committed:
adds router element

pzubar committed:
adjusts rendering of form elements in constructor

pzubar committed:
adjusts adding fields and options to form

pzubar committed:
adds ability to save form on server, improves ui a little bit

pzubar committed:
improves ui of form constructor, disables add field button after maximum of fields

pzubar committed:
adds simple homepage

pzubar committed:
changes API calls to firebase, adds ability to create and delete forms

pzubar committed:
fixes responsibility bug

pzubar committed:
adjusts some files to fit eslinter and prettier

pzubar committed:
formEditor became statefull component

pzubar committed:
adds form filler component

pzubar committed:
fixes several eslint errors

YakubchikV commented on src/actions/thunks.js:
```diff
@@ -0,0 +1,14 @@
+import { removeFormById } from '../models';
+import { showInfoMessage } from '../helpers/messages';
+
+export const requestDeleteForm = id => {
```
странные отступы здесь

YakubchikV commented on src/constants/index.js:
```diff
@@ -0,0 +1,3 @@
+export const API_PATH = 'http://forms-app.brutgroot.com/pzubar';
```
лучше записывать как //forms-app.brutgroot.com/pzubar'; - может в будущем быть https

YakubchikV commented on src/helpers/messages.js:
```diff
@@ -0,0 +1,6 @@
+import 'pnotify/dist/es/PNotifyButtons';
+import 'pnotify/dist/PNotifyBrightTheme.css';
+
+export const showErrorMessage = () => PNotify.error('An error occurred');
```
прокинь еще переменную, чтобы по message || PNotify.error('An error occurred');

YakubchikV commented on src/index.js:
```diff
@@ -1,25 +1,27 @@
+		<Router>
+			<Switch>
+				<Route exact path="/" component={HomePage} />
+				<Route path="/form/new" component={Editor} />
```
пути в константы, чтобы не менять в двух местах, а с одного можно было бы сменить везде

YakubchikV commented on src/middlewares/index.js:
```diff
@@ -0,0 +1,13 @@
+import { applyMiddleware, compose } from 'redux';
+import thunk from 'redux-thunk';
+
+const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
```
девтулз лучше добавлять когда у тебя не прод окружение, то есть можешь в скрипте добавить env переменную LOCAL=true чтобы включать какие-то фишки только в локальном окружении

YakubchikV commented on package.json:
```diff
@@ -4,28 +4,48 @@
   "description": "Final project in PDFfiller JS school",
```
добавить package-lock.json для фиксирования версий зависимостей, чтобы в один момент не упало ничего

YakubchikV commented on src/actions/thunks.js:
```diff
@@ -0,0 +1,14 @@
+		removeFormById(id).then(() => {
+			dispatch(deleteForm(id));
+			showInfoMessage('Form deleted');
+		});
```
catch где?

YakubchikV commented on src/pages/formEditor/components/FieldCard.jsx:
```diff
@@ -0,0 +1,76 @@
+import { FIELD_TYPES } from '../constants/fieldTypes';
+import { setFieldLabel, changeFieldType } from '../actions';
+
+const FieldCard = props => {
```
можно заюзать коннект без state6 первый параметр передать как null и тогда dispatch не нужно пробрасывать пропсом


function mapDispatchToProps(dispatch) {
  return {
    dispatch,
    ...bindActionCreators({ increment, decrement, reset }, dispatch)
  }

connect(
  null,
  mapDispatchToProps
)(Counter)

pzubar replied:
В этм случае решил отказаться от пробрасывания dispatch в "глупый" компонент и делать вместо этого логику в родительском "умном" компоненте

YakubchikV commented on src/pages/formEditor/selectors/index.js:
```diff
@@ -0,0 +1,4 @@
+export const getFields = state => state.formEditor.fields || {};
```
используй reselect

YakubchikV commented on src/pages/formFiller/components/FillCard.jsx:
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { Form, Input, Select } from 'semantic-ui-react';
+
+const FillCard = props => {
```
({ type, label, options })

pzubar committed:
adjusts hotfix for bug on form creation

pzubar committed:
fixes several eslint warnings and errors, improves form editor logic to not pass dispatch to dumb components

pzubar committed:
adds link to home from form editor component

YakubchikV commented on src/selectors/index.js:
```diff
@@ -0,0 +1,7 @@
+import { FORMS, FILLS } from '../constants';
```
это не селекторы, переделай на реселект

YakubchikV commented on src/pages/formEditor/thunks/index.js:
```diff
@@ -0,0 +1,42 @@
+import { addField, setName, setIsLoaded } from '../actions';
+import { addForm } from '../../../actions';
```
изза того, что у тебя две папки actions уже начинает возникать путаница

YakubchikV commented on src/pages/formEditor/thunks/index.js:
```diff
@@ -0,0 +1,42 @@
+import { addForm } from '../../../actions';
+import { showErrorMessage, showInfoMessage } from '../../../helpers/messages';
+import { getFormById, editFormById } from '../../../models';
+import { initialState } from '../reducer';
```
с редьюсерами тоже самое

pzubar committed:
fixes several PR notifications;
connects form editor element properly to redux according to optimisation issues
